### PR TITLE
Add support for ghc-8.10.1

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
Support for ghc-8.10.1 (at this time pulls the ghc-8.8.1-alpha2 branch).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -103,10 +132,7 @@ strategy:
     #   image: "vs2017-win2016"
     #   ghc-flavor: "da-ghc-8.8.1"
     #   resolver: "nightly-2019-09-26"
-    mac-da-ghc-881-881:
-      image: "macOS-10.13"
-      ghc-flavor: "da-ghc-8.8.1"
-      resolver: "nightly-2019-09-26"
+    # ---- ghc-flavor : da-ghc-8.10.1 (doesn't exist)
```
Comment (aherrmann-da) [inline azure-pipelines.yml]: Why is this one removed?

File: ghc-lib-gen/src/GhclibgenOpts.hs [hunk 2]
```diff
@@ -76,7 +76,8 @@ ghcFlavorOpt = option readFlavor
 
 readFlavor :: ReadM GhcFlavor
 readFlavor = eitherReader $ \case
+    "ghc-8.10.1" -> Right Ghc8101
     "ghc-8.8.1" -> Right Ghc881
     "da-ghc-8.8.1" -> Right DaGhc881
     "ghc-master" -> Right GhcMaster
-    flavor -> Left $ "Failed to parse ghc flavor " <> show flavor <> " expected ghc-8.8.1 or da-ghc-8.8.1"
+    flavor -> Left $ "Failed to parse ghc flavor " <> show flavor <> " expected ghc-8.8.1, da-ghc-8.8.1 or ghc-8.10.1"
```
Comment (aherrmann-da) [inline ghc-lib-gen/src/GhclibgenOpts.hs]: Why is `ghc-master` not listed as an expected input?

File: azure-pipelines.yml (hunk 1 above)
Comment (shayne-fletcher-da) [reply azure-pipelines.yml]: Argh. Just a mistake. Nice catch and thank-you. I don't know why but I struggle with that YAML format and keeping this matrix straight!

File: ghc-lib-gen/src/GhclibgenOpts.hs (hunk 2 above)
Comment (shayne-fletcher-da) [reply ghc-lib-gen/src/GhclibgenOpts.hs]: Good point. Just an oversight - I'll fix it. (actually, even then it's an approximation. You can pass a SHA in that position too)

**Pull request merged**